packages/ppx/src/Css_to_runtime.re

```diff
@@ -110,23 +110,29 @@ let rec render_at_rule = (~loc, at_rule: at_rule) => {
   };
 }
 and render_media_query = (~loc, at_rule: at_rule) => {
+  let (_, at_rule_prelude_loc) = at_rule.prelude;
   let parse_condition = {
-    let (value, at_rule_loc) = at_rule.prelude;
-    switch (value) {
-    | Variable(variable) => Ok(render_variable_as_string(variable))
-    | Paren_block([(Ident(_), ident_loc)]) =>
-      /* TODO: String.trim is a hack around, but a media query should be all clean. */
-      Ok(source_code_of_loc(ident_loc) |> String.trim)
-    /* In any other case, we believe on the source_code and transform it to string. This is unsafe */
-    | _whatever => Ok(source_code_of_loc(at_rule_loc) |> String.trim)
-    };
+    let prelude = source_code_of_loc(at_rule_prelude_loc) |> String.trim;
+    Css_property_parser.Parser.parse(
+      Css_property_parser.Parser.media_query_list,
+      prelude,
+    )
+    |> Result.map(_ => prelude);
   };
 
   let (delimiter, attrs) =
     Platform_attributes.string_delimiter(~loc=at_rule.loc);
 
   switch (parse_condition) {
-  | Error(error_expr) => error_expr
+  | Error(error_msg) =>
+    Error.expr(
+      ~loc=
+        Styled_ppx_css_parser.Parser_location.update_pos_lnum(
+          at_rule_prelude_loc,
+          loc,
+        ),
```

Comment:
When have you seen this to be needed? Can add a test for it? ^^

Reply:
this keeps the cnum of the first location but offset the lnum with the second loc. i've tried using intersection but this seems to work well to highlight the correct error location.